### PR TITLE
local cache for the results

### PR DESCRIPTION
**Pull request by volodymyrss**

### CODE REVIEW

File: oda_api/api.py [hunk 1]
```diff
@@ -1317,6 +1335,33 @@ def set_api_code(query_dict, url="www.astro.unige.ch/mmoda/dispatch-data"):
 
         return _cmd_
 
+
+    def save_result(self, response_json):
+        fn = self.unique_response_json_fn
+
+        os.makedirs(os.path.dirname(fn), exist_ok=True)
+
+        json.dump(response_json, gzip.open(fn, "wt"))
```
Comment (dsavchenko) [inline oda_api/api.py:1344]: Generally, I always prefer using `with` when opening file

File: oda_api/api.py [hunk 2]
```diff
@@ -1317,6 +1335,33 @@ def set_api_code(query_dict, url="www.astro.unige.ch/mmoda/dispatch-data"):
...
+        logger.info('saved result in %s', fn)
+
+
+    def load_result(self):
+        fn = self.unique_response_json_fn
+        logger.info('trying to load result from %s', fn)
+
+        t0 = time.time()
+        r = json.load(gzip.open(fn, 'rb'))
```
Comment (dsavchenko) [inline oda_api/api.py:1353]: I see it works (passes test) but it's not clear to me, why writing with `text` mode and reading with `bytes`

File: oda_api/api.py (hunk 1 above)
Comment (volodymyrss) [reply oda_api/api.py:1344]: yeah, well, it adds an extra line and an extra variable. Ultimately the file would get closed.
But ok, it's safer to make sure it's closed here.

File: oda_api/api.py (hunk 2 above)
Comment (volodymyrss) [reply oda_api/api.py]: it's just how `json` module works. Both text also works.

**Pull request merged**